### ISSUE TITLE
Error "simple protocol queries must be run with standard_conforming_strings=on" but standard_conforming_strings is on in the database.

### ISSUE THREAD

**Issue by OrvilleQ** [jackc/pgx#1368]
**Describe the bug**
Running app shows error `simple protocol queries must be run with standard_conforming_strings=on` but query database with `SHOW standard_conforming_strings;` returns `on`.

**To Reproduce**

1. Using neon.tech's serverless postgresql tech preview, psql version 14.

2. Running a custom version of headscale which could set `sslmode` to `verify-full` https://github.com/OrvilleQ/headscale/tree/fix-psql-sslmode.

3. `github.com/juanfont/headscale/cmd/headscale/cli/server.go:21 > Error initializing error="simple protocol queries must be run with standard_conforming_strings=on"`

**Expected behavior**
The simple protocol queries should work.

**Actual behavior**
It didn't.

**Version**
 - Go: `go version go1.19.2 linux/amd64`
 - PostgreSQL: `PostgreSQL 14.5 on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit`
 - pgx: `v4.17.2`

**Additional context**

**Comment by jackc** [comment]
When a connection is made to a PostgreSQL server it reports connection settings such as `standard_conforming_strings`. This is how pgx knows whether it is running with `standard_conforming_strings` or not. 

Given that this works when connecting to normal PostgreSQL directly I don't think this is a pgx bug. I suspect this is a bug where neon.tech or headscale is not reporting or is not passing through this message.

**Comment by ns-gzhang** [comment]
Just to add a relevant scenario I encountered recently. I was using `jackc/pgx/v5`, and the client connects to a proxy I developed using `jackc/pgx/v5/pgproto3`. Without fully understanding every bit of the protocol, the proxy didn't respond with `pgproto3.ParameterStatus` about `standard_conforming_strings` (it does not exist in the proxy and backend DB is not even a PG) and `client_encoding`. It worked perfectly with `psql`, but encountered the problems with the client without these settings. not sure if `psql` took the default? After adding these two ParameterStatus in response to the `StartupMessage`, the client worked, but took me quite a while to figure out :-)
